### ISSUE TITLE
Select - nested interactive elements

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7374]
### Expected Behavior
Automated accessibility tests shouldn't flag Select component as violating the nested interactive rule.

### Actual Behavior
We have a button with an input inside of it in the select and component. The input is needed to pass a value to a Form and the button is needed to open the select dropdown. Even though the input has tabIndex="-1" automated accessibility tools will still flag this as a nested interactive element. See more details here https://github.com/dequelabs/axe-core/issues/3227.

We could potentially explore a solution where we have a hidden input that is outside of the button, use this for the Form. Then just have text within the button where the input previously was to give the visual indication of what has been selected.

### URL, screen shot, or Codepen exhibiting the issue
https://storybook.grommet.io/?path=/story/input-select-simple--simple

### Steps to Reproduce

Go to the above story and look at the accessibility panel

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by NOMADE55** [comment]
I've been investigating this. I've found with a roadblock that might need some extra thought.
Grommet relies on user-agent's stylesheet min width for select inputs, as we do not define a size by default.

This doesn't fly with a span or div replacing the input's place., where we're going to need an specific value size if we want the content to fill as much as the input did previously.
![image](https://github.com/user-attachments/assets/ad23aec7-12c3-4eb0-baa3-c565de80a78e)

**Comment by jcfilben** [comment]
Good comments @NOMADE55, agree the input width makes this a bit tricky. Not sure at the moment how we might want to handle this but I'll post any updates here if I have any thoughts
